cleanrl/rnd_ppo.py

```diff
@@ -0,0 +1,1079 @@
+    parser.add_argument('--sticky-action', type=lambda x:bool(strtobool(x)), default=True, nargs='?', const=True,
+                          help='Toggles wheter or not to use sticky action.')
+
+    # RND arguments
+    parser.add_argument('--update-proportion', type=float, default=0.25)
+    parser.add_argument('--int-coef', type=float, default=1.0)
+    parser.add_argument('--ext-coef', type=float, default=2.0)
+    parser.add_argument('--int-gamma', type=float, default=0.99)
+    
+
+
+    args = parser.parse_args()
+    if not args.seed:
+        args.seed = int(time.time())
+
+
+args.batch_size = int(args.num_envs * args.num_steps)
+args.minibatch_size = int(args.batch_size // args.n_minibatch)
+
+class Flatten(nn.Module):
+    def forward(self, input):
+        return input.view(input.size(0), -1)
+
+class RNDModel(nn.Module):
+    def __init__(self, input_size, output_size):
+        super(RNDModel, self).__init__()
+
+        self.input_size = input_size
+        self.output_size = output_size
+
+        feature_output = 7 * 7 * 64
+
+        # Prediction network
+        self.predictor = nn.Sequential(
+            nn.Conv2d(
+                in_channels=1,
+                out_channels=32,
+                kernel_size=8,
+                stride=4),
+            nn.LeakyReLU(),
+            nn.Conv2d(
+                in_channels=32,
+                out_channels=64,
+                kernel_size=4,
+                stride=2),
+            nn.LeakyReLU(),
+            nn.Conv2d(
+                in_channels=64,
+                out_channels=64,
+                kernel_size=3,
+                stride=1),
+            nn.LeakyReLU(),
+            Flatten(),
+            nn.Linear(feature_output, 512),
+            nn.ReLU(),
+            nn.Linear(512, 512),
+            nn.ReLU(),
+            nn.Linear(512, 512)
+        )
+
+        # Target network
+        self.target = nn.Sequential(
+            nn.Conv2d(
+                in_channels=1,
+                out_channels=32,
+                kernel_size=8,
+                stride=4),
+            nn.LeakyReLU(),
+            nn.Conv2d(
+                in_channels=32,
+                out_channels=64,
+                kernel_size=4,
+                stride=2),
+            nn.LeakyReLU(),
+            nn.Conv2d(
+                in_channels=64,
+                out_channels=64,
+                kernel_size=3,
+                stride=1),
+            nn.LeakyReLU(),
+            Flatten(),
+            nn.Linear(feature_output, 512)
+        )
+
+        # Initialize the weights and biases
+        for p in self.modules():
+            if isinstance(p, nn.Conv2d):
+                nn.init.orthogonal_(p.weight, np.sqrt(2))
+                p.bias.data.zero_()
+
+            if isinstance(p, nn.Linear):
+                nn.init.orthogonal_(p.weight, np.sqrt(2))
+                p.bias.data.zero_()
```

Comment:
Consider using the `layer_init` function for consistency. I am not familiar with the architecture of RND, but shouldn't the output layer be initialize with `std=0.01`?

Reply:
In the original implementation, they used the sqrt(2).